app/code/core/Mage/Eav/Model/Entity/Abstract.php

```diff
@@ -993,18 +1051,52 @@ protected function _loadModelAttributes($object)
         foreach ($selectGroups as $selects) {
             if (!empty($selects)) {
                 $select = $this->_prepareLoadSelect($selects);
+                //reset order by because we sorting it in php to avoid filesort and tempoary table;
+                $select->reset(Zend_Db_Select::ORDER);
                 $values = $this->_getReadAdapter()->fetchAll($select);
+                uasort($values, function ($value1, $value2){
+                    if ((int)$value1['store_id'] == $value2['store_id']){
+                        return 0;
+                    }
+                    return $value1['store_id'] < $value2['store_id'] ? -1 : 1;
+
+                });
```

Comment:
Please see my suggestion for doing a quick grouping instead of a full sort: https://github.com/OpenMage/magento-lts/pull/347#issuecomment-334586482